vmc/resource_vmc_cluster_test.go

```diff
@@ -172,3 +174,48 @@ func testAccVmcClusterResourceImportStateIdFunc(resourceName string) resource.Im
 		return fmt.Sprintf("%s,%s", rs.Primary.ID, rs.Primary.Attributes["sddc_id"]), nil
 	}
 }
+
+func TestBuildClusterConfig(t *testing.T) {
+	type test struct {
+		input    map[string]interface{}
+		expected model.ClusterConfig
+	}
+
+	tests := []test{
+		{input: map[string]interface{}{
+			"num_hosts":          MinHosts,
+			"host_instance_type": HostInstancetypeI3,
+		}, expected: model.ClusterConfig{
+			NumHosts:         MinHosts,
+			HostInstanceType: String(model.SddcConfig_HOST_INSTANCE_TYPE_I3_METAL),
+		}},
+		{input: map[string]interface{}{
+			"num_hosts":          MinHosts,
+			"host_instance_type": HostInstancetypeI3EN,
+		}, expected: model.ClusterConfig{
+			NumHosts:         MinHosts,
+			HostInstanceType: String(model.SddcConfig_HOST_INSTANCE_TYPE_I3EN_METAL),
+		}},
+		{input: map[string]interface{}{
+			"num_hosts":          MinHosts,
+			"host_instance_type": HostInstancetypeI4I,
+		}, expected: model.ClusterConfig{
+			NumHosts:         MinHosts,
+			HostInstanceType: String(model.SddcConfig_HOST_INSTANCE_TYPE_I4I_METAL),
+		}},
+		{input: map[string]interface{}{
+			"num_hosts":          MinHosts,
+			"host_instance_type": HostInstancetypeR5,
+		}, expected: model.ClusterConfig{
+			NumHosts:         MinHosts,
+			HostInstanceType: String(model.SddcConfig_HOST_INSTANCE_TYPE_R5_METAL),
+		}},
+	}
+
+	for _, testCase := range tests {
+		var mockResourceSchema = schema.TestResourceDataRaw(t, clusterSchema(), testCase.input)
+		got, _ := buildClusterConfig(mockResourceSchema)
```

Comment:
You should not discard the error.
You should:
1. add a testcase that returns the error and check you got the correct one (so add an `expectedErr` field to your test struct
2. in all the other testcases, check that the returned error is `nil`


Also if the only thing that changes between tests is the instance type, write the tests to make that clear, like (untested):
```
	tests := []test{
		{instanceType: HostInstancetypeI3,
			expectedError: nil},
		{instanceType: HostInstancetypeI3EN,
			expectedError: nil},
		{instanceType: HostInstancetypeR5,
			expectedError: nil},
	}

	for _, testCase := range tests {
		config := map[string]interface{}{
			"num_hosts":          10,
			"host_instance_type": testCase.instanceType,
		}
                schema := schema.TestResourceDataRaw(t, clusterSchema(), config) // don't call it mock, there's no mock involved, you're calling all the real methods

		got, err := buildClusterConfig(&schema)
                assert.Equal(t, err, testCase.expectedError)
		if testCase.expectedError != nil {
                     // check error message
                }
	        assert.Equal(t, *got.HostInstanceType, testCase.instanceType)
	}
```